rust/src/storage/file/metadata_file.rs

```diff
@@ -0,0 +1,179 @@
+// Copyright (c) 2025 Sonic Operations Ltd
+//
+// Use of this software is governed by the Business Source License included
+// in the LICENSE file and at soniclabs.com/bsl11.
+//
+// Change Date: 2028-4-16
+//
+// On the date above, in accordance with the Business Source License, use of
+// this software will be governed by the GNU Lesser General Public License v3.
+
+use std::{
+    fs::File,
+    io::{Read, Seek, SeekFrom, Write},
+};
+
+use crate::storage::Error;
+
+/// Metadata stored in the metadata file.
+#[derive(Debug)]
+pub struct Metadata {
+    pub node_count: u64,
+    pub reuse_frozen_count: u64,
+}
+
+/// A file that stores metadata about the node store for a certain node type.
```

Comment:
I found two other comments referencing a "node store", but it does not seem to really be a well-defined term (i.e., there's no type called `NodeStore` or similar).